src/Uno.UI/UI/Xaml/DependencyObjectExtensions.cs

```diff
@@ -440,8 +440,10 @@ internal static void InvalidateArrange(this DependencyObject d)
 
 		internal static void InvalidateRender(this DependencyObject d)
 		{
-			var uielement = d as UIElement ?? d.GetParents().OfType<UIElement>().FirstOrDefault();
-			uielement?.InvalidateRender();
+			// TODO Uno: This currently doesn't work as expected since there is no UIElement.InvalidateRender method.
```

Comment:
From the WinUI source, there's an `InvalidateRender()` method on `CalendarBaseViewItem` but none on `UIElement`. I notice that the CalendarViewBaseItem method was imported, but marked private.

I would suggest the following fix:
 - make `CalendarViewBaseItem.InvalidateRender()` internal
 - delete this extension method entirely, to avoid similar bugs in the future

Reply:
@davidjohnoliver That's A LOT better! Thanks 🎉